runcoms/zprofile

```diff
@@ -66,11 +66,11 @@ fi
 # Temporary Files
 #
 #
-if [[ -z "$TMPDIR" ]]; then
-  export TMPDIR="/tmp/zsh-$UID"
+# Set TMPDIR if the variable is not set or the directory doesn't exist
+if [[ ! "${TMPDIR}" ]]; then
```

Comment:
`-z` works fine. `[[ ! "$var" ]]" is excatly equivalent to `[[ -z "$var ]]`. I can change it back though. I find the ! clearer but that's just me. I can change it to `-z` if people are happier.